src/axl_pthread.c

```diff
@@ -283,10 +288,11 @@ int axl_pthread_test (int id)
 {
     kvtree* file_list = kvtree_get_kv_int(axl_file_lists, AXL_KEY_HANDLE_UID, id);
     int64_t ptr;
-    struct axl_pthread_data *pdata;
+    struct axl_pthread_data *pdata = NULL;
 
     kvtree_util_get_int64(file_list, AXL_KEY_PTHREAD_DATA, &ptr);
     pdata = (struct axl_pthread_data *) ptr;
+    assert(pdata);
```

Comment:
Can you explain these few lines to me (L291-L295)? When does this assert fail?

Reply:
It's there for sanity checking only.  It could be useful to catch if someone does an `AXL_Test()` before the pthread data is initialized.

Reply:
Hmmm. We should probably define the behavior of AXL if you call test before dispatching (or is it initializing?) a transfer. We probably don't want to die. I think there is a more graceful/helpful error response we could do here.

Reply:
Nevermind, looks like we're currently sanity checking AXL_Test to only check during a dispatched transfer.
```
int AXL_Test (int id)
{
...
    /* check that handle is in correct state to test */
    if (xstate != AXL_XFER_STATE_DISPATCHED) {
        AXL_ERR("Invalid state to test UID %d", id);
        return AXL_FAILURE;
    }
```